### ISSUE TITLE
Please add "Discussions" 

### ISSUE THREAD

**Issue by RezaRahmati** [nestjs/nest#7741]
* Discord Community Chat: https://discord.gg/G7Qnnhy
* StackOverflow: https://stackoverflow.com/questions/tagged/nestjs using the tag `nestjs`
* Twitter: If it's just a quick question you can ping our Twitter: https://twitter.com/nestframework

Hi @kamilmysliwiec 
I have seen a lot of question topics that mainly are closed and people are referred to use discord, the issue is, the knowledge generated in discord is not searchable, another dev might have same question and discord support team needs to answer same question over and over.

Today I wanted to send a question on another repo and they had solved this issue pretty good. if you click on "Question and help" on this [link](https://github.com/cypress-io/cypress/issues/new/choose), it takes you to *Discussions*, which is pretty good, and are not mixed with bug reports.

It would be great idea to enable this feature.

What do you think?

**Comment by jmcdo29** [comment]
We've floated the idea of discussions a few times with other members of the team. I think the eventual goal is to have both Discord and Discussions as an option. The difficult part there is how do we decide what needs to be a discussion, what needs to be a Discord conversation, and how do we make users aware of both possibilities. 

Personally, I'm all for discussions, I'm using them for some of my other open source libraries and love how they function. I think the searchability will bring a lot of value for those looking for answers.

**Comment by hedgerh** [comment]
Has answering the same questions over and over on Discord been a pain point for the team?

I'd be interested in first hearing how Discord support has been going and if there are pain points, then figuring out what the best solution or solutions may be.

If there are common questions being asked, the best solution may be to add/change documentation, for example.  Now that I think about it, the two questions I've asked in Discord could probably have best been addressed with docs.

P.S. Just wanna say that the support I've received on Discord has been amazing.  It's made me want to give back however I can.

**Comment by RezaRahmati** [comment]
> Has answering the same questions over and over on Discord been a pain point for the team?
> 
> I'd be interested in first hearing how Discord support has been going and if there are pain points, then figuring out what the best solution or solutions may be.
> 
> If there are common questions being asked, the best solution may be to add/change documentation, for example. Now that I think about it, the two questions I've asked in Discord could probably have best been addressed with docs.
> 
> P.S. Just wanna say that the support I've received on Discord has been amazing. It's made me want to give back however I can.

The point is your questions might have been some other devs question as well, the knowledge in the discord is not searchable, when you stuck in something you start googling, if the q&a exists on the web it's very possible you can solve your issue without need to contact support team on discord

**Comment by jmcdo29** [comment]
I'll admit, I end up answering the same question _a lot_, not matter how clearly we think the docs state it. Someone, somehow, always misinterprets the writing and the examples.

The biggest pain point, I think is what Reza brought up, searchability. Not being able to search for past issues and seeing the result of the conversation. The second biggest issue, in my opinion, is blaming the framework for an issue when it's really the underlying library that is a problem. 

Personally, I've loved using Discord for support, it's quick and easy, the formatting support is great, and now that threads are added the main chats stay a lot cleaner, which is really nice. But in the end, the answers aren't searchable. I think the use of Discord is great when it comes to discussions about how to implement advanced functionality and bouncing ideas off of other devs, but I do see the draw of using GitHub Discussions for Knowledge Sharing.

**Comment by RezaRahmati** [comment]
> We've floated the idea of discussions a few times with other members of the team. I think the eventual goal is to have both Discord and Discussions as an option. The difficult part there is how do we decide what needs to be a discussion, what needs to be a Discord conversation, and how do we make users aware of both possibilities.
> 
> Personally, I'm all for discussions, I'm using them for some of my other open source libraries and love how they function. I think the searchability will bring a lot of value for those looking for answers.

True, it's hard to direct people to multi channels, but people will find their ways, like now we post on stack overflow and if it's urgent we reach to discord

**Comment by hedgerh** [comment]
Searchability is beneficial in reducing the support load on the team, but I don't think it improves a user's experience. 

As a user, would you rather:
- Search through past issues that may or may not answer your question 
- Just ask your question on Discord and quickly get live support

If you don't care about burdening the support team, the second option seems much more convenient, imo.

If you _do_ care about burdening the support team, or we're looking for ways to make their lives easier, that's a different story.

Hopefully that makes sense and is helpful!  I just think, as a user, live support beats crawling google/github/SO for answers anyday.

In general, I'm 1000% 👍 on GitHub Discussions, though.

**Comment by kamilmysliwiec** [comment]
I think we discussed this a few months ago (when Github Discussions feature was released) but I cannot find a link to this issue.

Currently, we're using Discord as our primary communication platform. In the past (before Discord), we were using Gitter and so we already have 1 migration behind us. Now, when we have a strong & huge community on Discord, with almost 20k members we definitely don't plan to move away from it in the foreseeable future.

Now, why can't we have both (Github Discussions and Discord) at the same time? Although I'd love to have the Discussions tab available in the nest repository with plenty of topics covered, with the searchability it provides, etc., this is simply impossible to maintain. Another thing to manage would unfortunately drastically reduce the time we have for maintaining the actual framework & codebase and the whole ecosystem. If you analyze "Issues" tab, you'll very likely draw two conclusions:
- most people tend to ask questions irrelevant to the framework, reporting "bugs" that usually come from their projects (wrong configuration, missing setup step, etc.)
- in most issues, you'll find myself & @jmcdo29 answering to people's problems (kudos to everyone else who sometimes chimes in!) and that's it

At this point, my every day starts with going through hundreds of notifications from Github. Adding another platform on top of the existing Discord channel (which works great, thanks for the community & and @jmcdo29 who is incredibly active there) + Stackoverflow doesn't make sense. This indeed may make sense for smaller libraries but for bigger frameworks with established communication platforms, it doesn't (and this is probably the reason why other popular frameworks don't have this feature either). Not to even mention how we would decide whether something should be discussed on Github Discussion or Discord, where to direct people, and so on, and so forth.
